test/sagemaker_tests/pytorch/training/conftest.py

```diff
@@ -493,6 +493,68 @@ def skip_s3plugin_test(request):
             )
 
 
+@pytest.fixture(autouse=True)
+def skip_smdebug_test(
+    request,
+    processor,
+    ecr_image,
+):
+    if request.node.get_closest_marker("skip_smdebug_test"):
+        _, image_framework_version = get_framework_and_version_from_tag(ecr_image)
+        image_cuda_version = get_cuda_version_from_tag(ecr_image) if processor == "gpu" else ""
+        if (
```

Comment:
there seems to be quite a bit of repeated code in this file. Recommend creating a function to wrap repeated code

Reply:
I agree that there a lot of repeated code here but that is also because the current skip conditions are the same. TIt is also likely for the if conditions to change depending on what package we add to each DLC.

These skip tests for each package is the result of our DLC image size reduction when we removed all unnecessary packages. And now that our releases are decoupled from a few partner teams, we'll have these specific skip conditions when doing currency release which is not ideal.

Initially the skip conditions are done per framework eg: `skip_pt21_test` `skip_pt22_test`. But if we are doing that going forward, we'll end up with a long list of framework to skip, which is why I have consolidated those into skipping tests per related packages.